eng/tools/generate-dev-builds/index.js

```diff
@@ -0,0 +1,218 @@
+const path = require("path");
+const util = require("util");
+const fs = require("fs");
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
+const parse = require(`${__dirname}/../../../common/lib/jju/parse`).parse;
+const semver = require('semver');
+
+const appendPackageData = (data, pkgSrc, pkgJson) => {
+
+  var dep = new Map();
+  for(var key in pkgJson.dependencies){
+    dep.set(key, pkgJson.dependencies[key]);
+  }
+
+  var devDep = new Map();
+  for(var key in pkgJson.devDependencies){
+    devDep.set(key, pkgJson.devDependencies[key]);
+  }
+
+  data[pkgJson.name] = {
+    src: pkgSrc,
+    ver: pkgJson.version,
+    dep: dep,
+    devDep: devDep,
+    visited: false,
+    visitedDep : false
+  };
+};
+
+const getRushPackages = async (rushPath) => {
+  const baseDir = path.dirname(rushPath);
+  const rushJson = parse(await readFile(rushPath, "utf8"));
+  const packageData = {};
+
+  for (const proj of rushJson.projects) {
+    const projDir = path.join(baseDir, proj.projectFolder);
+    const packageJson = parse(await readFile(path.join(projDir, "package.json"), "utf8"));
+    appendPackageData(packageData, projDir, packageJson);
+  }
+
+  return packageData;
+};
+
+async function writeToPackageJson(pckJsonPath,contents){
+  console.log("Inside writeTo packageJson function for - " + pckJsonPath );
+  var data = JSON.stringify(contents,null,2);
+  try{
+      fs.access(pckJsonPath, fs.F_OK, (err) => {
+      if (err) {
+        console.error(err);
+        return;
+      }
+    });
+
+    await fs.writeFile(pckJsonPath, data, (err) => {
+      if(err){
+        console.error(err);
+        return;
+      }
+    });
+    console.info("File " + pckJsonPath + " created successfully with Node.js v10 fs/promises!");
+  }
+  catch(e){
+    console.error(`\n\n${e.toString()}\n\n`);
+    process.exit(1);
+  }
+  
+}
+
+function editPackageJson(rushPackages){
```

Comment:
Do you know if there are any rush or npm version commands that can update the package version for you instead of needing to maintain code to read and write the project files manually?

Reply:
There is npm version command for updating the version in package.json - but I will need to run that from command line and from within the folder of the respective package. And for updating versions of dependencies if we will still do it, then I will need to update them manually

Reply:
There is also https://rushjs.io/pages/commands/rush_version/ but not sure if we have to buy into their proscribed versioning strategy if we use this.